spec/rspec/rails/example/view_example_group_spec.rb

```diff
@@ -151,19 +151,19 @@ def _default_file_to_render; end # Stub method
         it "converts the filename components into render options" do
           allow(view_spec).to receive(:_default_file_to_render) { "widgets/new.en.html.erb" }
           view_spec.render
-          expect(view_spec.received.first).to eq([{template: "widgets/new", locales: ['en'], formats: [:html], handlers: ['erb']}, {}, nil])
+          expect(view_spec.received.first).to eq([{template: "widgets/new", locales: [:en], formats: [:html], handlers: [:erb]}, {}, nil])
         end
 
         it "converts the filename with variant into render options" do
           allow(view_spec).to receive(:_default_file_to_render) { "widgets/new.en.html+fancy.erb" }
           view_spec.render
-          expect(view_spec.received.first).to eq([{template: "widgets/new", locales: ['en'], formats: [:html], handlers: ['erb'], variants: ['fancy']}, {}, nil])
+          expect(view_spec.received.first).to eq([{template: "widgets/new", locales: [:en], formats: [:html], handlers: [:erb], variants: [:fancy]}, {}, nil])
         end
 
         it "converts the filename without format into render options" do
           allow(view_spec).to receive(:_default_file_to_render) { "widgets/new.en.erb" }
           view_spec.render
-          expect(view_spec.received.first).to eq([{template: "widgets/new", locales: ['en'], handlers: ['erb']}, {}, nil])
+          expect(view_spec.received.first).to eq([{template: "widgets/new", locales: [:en], handlers: [:erb]}, {}, nil])
```

Comment:
I would feel confident with merging this to `main` if we had a confirmation that with `to_sym` all supported Rails versions (from soft-supported 5.0 and 5.1) up to 6.1 still work fine.

Can you recall, what was the failing cucumber scenario for 7.0 without `.to_sym` exactly?

Reply:
I can't recall but I think that calling `render` without arguments raises `ActionView::MissingTemplate`.

Reply:
It seems that an exception is raised if the path described in the description contains handler, locales, etc. as shown below. Sorry I don't know that there is such a test in rspec-rails' cucumber.

```ruby
RSpec.describe "root/index.html.erb", type: :view do # also xxx.html.slim, xxx.en.html.erb 
  it { render } #=> raise ActionView::MissingTemplate
end
```

Reply:
My concern is that the change would break on an older version of Rails. As this is something that was not thought forward to be changed in the future, there might not be a test for it. Sorry for being extra cautious.

Reply:
@pirj In that case would it be possible to release this as a new major version

Reply:
Only if we drop support for all versions that have a chance not to support this. In `rspec-rails` 6 we will most probably still keep support for Rails 6.0, and this means we have to make sure the approach with symbols works for it, too.

This is purely theoretical. I have a gut feeling that with Symbols it will work with 5.2 and probably even [earlier versions](https://github.com/rspec/rspec-rails/pull/2115/files#diff-104fd0e14717644c4d9887ea03b002a615a7f41020003a78ab6992fa425e7beeR131), too. But to be on the same side, I'd love some confirmation that `requested.handlers` in Rails 6.0 would accept a Symbol.
Otherwise, we'll have to add a conditional Rails version check in the runtime and [specs](https://github.com/rspec/rspec-rails/pull/2115/files#diff-a5c90dfedaef6ce2253b56400927211db833791f14651c2aab7b7a33df40cb15R140).

Frankly speaking, I don't have a clear plan how to proceed with this. @JonRowe ?
#2537 is standing in our way to get a green Rails 7 build. I hope to get to it next week.

Reply:
@pirj So I just saw this discussion after making a similar commit to the PR I opened a few days ago.  I'm a little confused as to what the concern is.

We know symbols work on Rails versions in CI because the relevant Cucumber feature (`features/view_specs/view_spec.feature:124`) passes for those versions.  If symbols didn't work on a particular Rails version that feature would fail.

Reply:
> features/view_specs/view_spec.feature:124

Fair enough. For some reason, I failed to find this feature.

Wondering if it works equally fine with `variant` and `locale`.

For the reference, a [discussion about `format.to_sym`](https://github.com/rspec/rspec-rails/pull/1452).
This discussion has an argument "if they don't support symbols, it's a Rails bug, and people using older versions of Rails, they should stick to older versions of `rspec-rails`". We've started adhering to semver since then, and currently we officially support Rails 5.2 and soft-support 5.0 and 5.1 (please correct if I'm wrong with this again). From my perspective, it would be quite irresponsible to break the current `rspec-rails` 5.0.x for users of Rails 5.
Even though we claim that:
> According to RSpec Rails new versioning strategy use:
>    rspec-rails 5.x for Rails 6.x.
>    rspec-rails 4.x for Rails from 5.x or 6.x.

and 
> ### 5.0.0 / 2021-03-09
> * Drop support for Rails below 5.2.

We have a release strategy @JonRowe has described [here](https://github.com/rspec/rspec-rails/pull/2546#issuecomment-1009785284) that I've recently violated with the merge of that PR, hopefully without any breaking impact for the users.
For the future researchers, it might be surprising that Rails 7.0 compatibility support appears in `rspec-rails` 5.0.3/5.1.0 releases that target Rails 6.0 and 6.1.

I suggest branching `rails-7-0-dev` from `main`, and merging this PR and #2552 to it.
When we release 5.0.3 and 5.1.0, it can be merged back to `main`, and it will be time for `rspec-rails` 6.0.

@JonRowe Does this make sense?

Reply:
For what it's worth, that approach sounds reasonable to me.

That said I'm not sure why there's a "soft support" for Rails 5 and 5.1.  If you want to support them, why not just add them to CI?  If they can't be added to CI, are they actually supported?

It's worth noting that PR #2552 + #2546 has only 2 code changes of very limited scope that could end users of spec-rails:

1. The symbol change for the handler argument ("erb" becomes :erb before being passed to ActionView::Base).  This only becomes relevant in a particular type of view spec.
2. The changes to argument handling for Action Mailer matches (in my view this is the far more likely to break item between Rails versions)

On top of these changes, this PR symbolizes a few other values passed to ActionView::Base.

That's a very small, tightly scoped set of changes.  So the potential blast radius and risk of impact on end users is very limited.

Reply:
> if you want to support them, why not just add them to CI

We don't want to support them, and we don't want to break them. The reason is to drop some burden, mostly branching in specs:
 - [drop support for Rails < 5.2](https://github.com/tagliala/rspec-rails/commit/686bf845ad17704b757aafdd8388058563546b0c)
 - [drop support for Rails < 5.0](https://github.com/rspec/rspec-rails/pull/2226)

Actually, by looking at the former, there are [two changes](https://github.com/tagliala/rspec-rails/commit/686bf845ad17704b757aafdd8388058563546b0c#diff-cf41d523f2c21002e68799050ceb4c8e781e1d291680537457a3bebdf801161bL10) that presumably break Rails 5.0 support. But 5.1 may still work.

>  PR #2552 + #2546 has only 2 code changes of very limited scope

Right 👍
Please don't get me wrong with my rant above, I'm all for merging both ASAP and releasing this as 6.0.0-pre without bothering with 5.0.3 and 5.1.0.

Reply:
I guess I just consider "soft support" very harmful because it adds ambiguity and discussion to almost any change, without any kind of authoritative answer.  If it's in CI, then you know you didn't break it (or you broke it and need to fix it).  Saving a few branches is in my view not worth the ambiguity on nearly every material PR.  In my experience it sows up work (and open source work) substantially.

But I'm not a maintainer here, so I'll of course defer to the library maintainers.